app/controllers/api/items_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::ItemsController < Api::BaseController
+  def fetch_latest_products
+    page = params[:page].to_i
+    limit = params[:limit].to_i
+    offset = (page - 1) * limit
+
+    query = Item.joins(:collection).select(
+      'items.id, items.name, items.price, items.discounted_price, collections.name as collection_name'
+    ).order(id: :desc)
+
+    query = query.limit(limit).offset(offset) if page > 0 && limit > 0
```

Comment:
The conditional pagination logic is good, but ensure that the 'page' and 'limit' parameters are validated for proper integer values before using them to avoid potential errors or injection attacks.